addons/website_blog/views/website_blog_templates.xml

```diff
@@ -133,36 +133,34 @@
                 </div>
 
                 <div t-foreach="blog_posts" t-as="blog_post" class="mb32">
-                  <span t-field="blog_post.author_avatar" class="o_not_editable" t-field-options='{"widget": "image", "class": "img-circle pull-right mt16 media-object"}' />
-                  <h2>
-                    <a t-attf-href="/blog/#{ slug(blog_post.blog_id) }/post/#{ slug(blog_post) }" t-field="blog_post.name" class="mb4"/>
-                  </h2>
-
-                  <div class="text-muted">
-                      <h4
-                        t-field="blog_post.subtitle"
-                        class="mb4 mt4"/>
-                    <div name='blog_post_data' class='mb0'>
-                         <span t-field="blog_post.author_id" style="display: inline-block;" t-field-options='{
+                    <t t-if="has_author_avatar(blog_post.author_id.id)">
```

Comment:
Test on `if blog_post.author_avatar:`, since this field will be false if no image is defined.